packages/aws-cdk-lib/aws-stepfunctions/lib/activity.ts

```diff
@@ -15,6 +17,26 @@ export interface ActivityProps {
    * @default - If not supplied, a name is generated
    */
   readonly activityName?: string;
+
+  /**
+   * Specifies a symmetric customer managed KMS key for server-side encryption of the activity inputs.
+   * Step Functions will reuse the key for a maximum of `kmsDataKeyReusePeriodSeconds`.
+   *
+   * @default - data is transparently encrypted using an AWS owned key
+   */
+  readonly kmsKey?: kms.IKey;
+
+  /**
+   * Maximum duration that Step Functions will reuse customer managed data keys.
+   * When the period expires, Step Functions will call GenerateDataKey.
+   *
+   * You can only provide a value if `kmsKey` is set.
+   *
+   * Must be between 60 and 900 seconds.
+   *
+   * @default Duration.seconds(300)
+   */
+  readonly kmsDataKeyReusePeriodSeconds?: Duration;
```

Comment:
Because this property is dependent on whether `kmsKey` is set, what you can do instead of exposing both fields is to create a single optional property that takes in a KmsKeyConfig interface that contains both `kmsKey` and `kmsDataKeyReusePeriodSeconds` like so:
```ts
readonly kmsKeyConfig?: KmsKeyConfig;

export interface KmsKeyConfig: {
  readonly kmsKey: kms.IKey;
  readonly kmsDataKeyReusePeriodSeconds?: Duration;
}
```
Now both properties are still optional, but it is strictly enforced that`kmsDataKeyReusePeriodSeconds` can only be set when `kmsKey` is too.

Reply:
I believe it was structured in this way to follow this guideline: https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#flat

Reply:
Ah good point, you're absolutely right. There are specific cases where if the construct is not deeply nested we would make an exception to that but in this case it wouldn't be ideal. Regardless though we don't want to have the user keep track of when or when not to set the property. 

What we could do instead here is make the field take in a class and have that class be something like

```ts
class KmsConfig {
  public static setKmsKeyProps(key: IKey, resuseSeconds?: Duration) {
    return new KmsConfig(key, reuseSeconds);
  }

  constructor(public readonly key, public readonly reuseSeconds);
}
```
And then we would use that to plug the correct values in the correct place. This way, if more related fields are ever created, we can either add more static functions for those settings or se can adjust the current one.

Reply:
Can we just use the constructor instead of a static method? 

Using the same names as [EncryptionConfiguration](https://docs.aws.amazon.com/step-functions/latest/apireference/API_EncryptionConfiguration.html) in the Step Functions API to follow the naming recommendation in the design guideline: https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#naming

```ts
class EncryptionConfiguration {
    constructor(public readonly kmsKey: kms.IKey, public readonly kmsDataKeyReusePeriodSeconds?: Duration) {
        this.kmsKey = kmsKey;
        this.kmsDataKeyReusePeriodSeconds = kmsDataKeyReusePeriodSeconds;
    }
}
```

In use, it would be:
```ts
const kmsKey = new kms.Key(this, 'Key');
const stateMachine = new sfn.StateMachine(this, 'StateMachineWithCMKEncryptionConfiguration', {
  /* ... other props ... */
  encryptionConfiguration: new EncryptionConfiguration(kmsKey, cdk.Duration.seconds(60))
});
```

Or, omitting the optional data key reuse period:
```ts
const kmsKey = new kms.Key(this, 'Key');
const stateMachine = new sfn.StateMachine(this, 'StateMachineWithCMKEncryptionConfiguration', {
  /* ... other props ... */
  encryptionConfiguration: new EncryptionConfiguration(kmsKey)
});

Reply:
Yes that should be fine. The naming convention is good too.